x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/MountSearchableSnapshotRequest.java

```diff
@@ -38,7 +38,7 @@
 public class MountSearchableSnapshotRequest extends MasterNodeRequest<MountSearchableSnapshotRequest> {
 
     public static final ConstructingObjectParser<MountSearchableSnapshotRequest, RestRequest> PARSER = new ConstructingObjectParser<>(
-        "mount_searchable_snapshot", true,
+        "mount_searchable_snapshot", false,
```

Comment:
Can we please revert this to `true` again? I understand having the parser rejecting unknown fields would have helped to catch the wrongly formatted request body in the first place but moving this to `false` means that it will reject any new fields in the future. 

We kind of rely on ignoring unknown fields to not break compatibility when the High Level REST Client and the server are not in the exact same version (https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-compatibility.html).

We're working on a better version support for REST APIs that could avoid this leniency in the future. I also  checked the new official Java Client for Elasticsearch (https://github.com/elastic/elasticsearch-java/) and it does use the right `ignore_index_setting`.

Reply:
Thanks for your reivew, I've reverted the parameter to `true`, can you help to take a look?